### ISSUE TITLE
[en] A sequence of underscores causes a spelling warning

### ISSUE THREAD

**Issue by MikeUnwalla** [languagetool-org/languagetool#1204]
In the same way that LT ignores a sequence of hyphens, it should ignore a sequence of underscores. Or better, have a style rule that says, "Make sure that this sequence of underscores is correct."

![image](https://user-images.githubusercontent.com/5816160/46275410-c94d2100-c554-11e8-914b-79972d61f0c8.png)

**Comment by SkyCharger001** [comment]
Personally I use underlined spaces for this instead of underscores, as it allows one to include 'blanks' that are already filled in. (EG: a requisition form with four items, while the form allows for twelve ... with a typed font for the pre-printed and a script font for the items to further sell the idea that it's a filled in form.)

**Comment by f-knorr** [comment]
You're right! This behaviour is directly related to my change which has introduced _ as a word boundary (before change: mid_term = 1 token; after change: mid_term = 3 tokens). In German and probably several other languages, this has been the default behaviour for quite some time and there haven't been any complaints.
I have chosen this change as it is more consistent regarding German and other languages. Another approach would have been to introduce a disambiguation rule that makes LT ignore the spelling of sequences of _. 
I do not know what solution is best. You could, of course, write another rule that raises an error for `[a-z]+_[a-z]+` sequences...

**Comment by SkyCharger001** [comment]
This is just my view, but:
I think this change should be allowed to stay, as underscores are a common way of denoting non-breaking-spaces in text encodings that don't have native NBS support.

**Comment by MikeUnwalla** [comment]
@f-knorr wrote: You could, of course, write another rule that raises an error...
OK, I will do that.

**Comment by baarsrj** [comment]
I was very surprised to see this! Indeed, an underscore is not a valid word char, but it is not a valid separator too; just junk. So I would expect it to raise a spell checking error. But I will add a rule for it too.
